extensions/amp-ad/0.1/amp-ad-ui.js

```diff
@@ -198,6 +201,49 @@ export class AmpAdUIHandler {
     this.baseInstance_.element.appendChild(uiComponent);
     return uiComponent;
   }
+
+  /**
+   * @param {number|string|undefined} height
+   * @param {number|string|undefined} width
+   * @param {number} iframeHeight
+   * @param {number} iframeWidth
+   * @return {!Promise<Object>}
+   */
+  updateSize(height, width, iframeHeight, iframeWidth) {
+    // Calculate new width and height of the container to include the padding.
+    // If padding is negative, just use the requested width and height directly.
+    let newHeight, newWidth;
+    height = parseInt(height, 10);
+    if (!isNaN(height)) {
+      newHeight = Math.max(this.element_./*OK*/offsetHeight +
+          height - iframeHeight, height);
+    }
+    width = parseInt(width, 10);
+    if (!isNaN(width)) {
+      newWidth = Math.max(this.element_./*OK*/offsetWidth +
+          width - iframeWidth, width);
+    }
+
+    const sizes = {
+      newWidth,
+      newHeight,
+    };
+
+    if (!newHeight && !newWidth) {
+      return Promise.reject(sizes);
+    }
+
+    if (getAdContainer(this.element_) == 'AMP-STICKY-AD') {
+      // Special case: force collapse sticky-ad if no content.
+      return Promise.reject(sizes);
```

Comment:
This is a bit tricky. Since this doesn't call `#attemptChangeSize`, we don't go through the `#overflowCallback` flows, which could leave the ad creative in a weird state.